src/orion/core/worker/primary_algo.py

```diff
@@ -152,6 +152,19 @@ def suggest(self, num: int) -> list[Trial]:
                         f"Space: {self.transformed_space}"
                     )
                 original = self.transformed_space.reverse(transformed_trial)
+                if transformed_trial.parent is not None:
```

Comment:
Assuming that `trial.parent: Optional[str]`:
```suggestion
                if transformed_trial.parent:
```